homeassistant/components/homeassistant_sky_connect/__init__.py

```diff
@@ -24,19 +39,44 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         # The USB dongle is not plugged in
         raise ConfigEntryNotReady
 
-    usb_info = usb.UsbServiceInfo(
-        device=entry.data["device"],
-        vid=entry.data["vid"],
-        pid=entry.data["pid"],
-        serial_number=entry.data["serial_number"],
-        manufacturer=entry.data["manufacturer"],
-        description=entry.data["description"],
-    )
+    addon_manager: AddonManager = get_addon_manager(hass)
+    try:
+        addon_info: AddonInfo = await addon_manager.async_get_addon_info()
+    except AddonError as err:
+        _LOGGER.error(err)
+        raise ConfigEntryNotReady from err
+
+    # Start the addon if it's not started
+    if addon_info.state == AddonState.NOT_RUNNING:
+        await addon_manager.async_start_addon()
+
+    if addon_info.state not in (AddonState.NOT_INSTALLED, AddonState.RUNNING):
+        _LOGGER.debug(
+            "Multi pan addon in state %s, delaying yellow config entry setup",
+            addon_info.state,
+        )
+        raise ConfigEntryNotReady
+
+    if addon_info.state == AddonState.NOT_INSTALLED:
```

Comment:
Is this the source of truth if the user wants multi-pan or not? I think that we should store it in the config entry options as HA is the source of truth. And it should setup the add-on correctly based on that.

Reply:
No, that was indeed not correct. Updated so we check if the addon is configured to use the USB adapter belonging to this config entry 👍